src/firefly/js/ui/SwitchInputField.jsx

```diff
@@ -62,7 +68,9 @@ SwitchInputField.propTypes = {
     fieldKey: PropTypes.string.isRequired,
     groupKey: PropTypes.string,
     initialState: PropTypes.shape({
-        value: PropTypes.bool
+        value: PropTypes.any // changed from bool → any to support string values
     }),
+    trueValue: PropTypes.any,
+    falseValue: PropTypes.any,
```

Comment:
I don't think any of this is necessary. The state of switch component will always be binary. The consumer components should translate string that they want to true/false.